features/src/main/feature/camel-features.xml

```diff
@@ -2066,12 +2066,14 @@
         <feature version='[5,6)'>http-client</feature>
         <bundle>mvn:org.apache.camel.karaf/camel-workday/${project.version}</bundle>
     </feature>
-<!--    <feature name='camel-xchange' version='${project.version}' start-level='50'>-->
-<!--        <feature version='${project.version}'>camel-core</feature>-->
-<!--        <bundle dependency='true'>wrap:mvn:org.knowm.xchange/xchange-core/${xchange-version}</bundle>-->
-<!--        <bundle dependency='true'>wrap:mvn:com.github.mmazi/rescu/${rescu-version}</bundle>-->
-<!--        <bundle>mvn:org.apache.camel/camel-xchange/${project.version}</bundle>-->
-<!--    </feature>-->
+    <feature name='camel-xchange' version='${project.version}' start-level='50'>
+        <feature version='${project.version}'>camel-core</feature>
+        <feature>jackson</feature>
```

Comment:
Doesn't seem to be used/needed

Reply:
it could be removed for compilation, but I doubt that it works at runtime since there are many dependencies of xchange-core that relates to jackson packages, but they are optional by default in the wrap protocol.

Reply:
Are-you sure that it is used at runtime? Because according to the command `mvn dependency:tree -Dscope=runtime` in the camel project, there is no such dependency:

Reply:
Not sure they are used at runtime but probably since they are used in the dto packages, this is how I saw them :

```
karaf@root()> install wrap:mvn:org.knowm.xchange/xchange-core/5.1.1
Bundle ID: 107
karaf@root()> headers 107

Export-Package = 
        org.knowm.xchange;
                uses:="org.knowm.xchange.client,
                        org.knowm.xchange.dto.meta,
                        org.knowm.xchange.exceptions,
                        org.knowm.xchange.instrument,
                        org.knowm.xchange.service.account,
                        org.knowm.xchange.service.marketdata,
                        org.knowm.xchange.service.trade,
                        org.slf4j,
                        si.mazi.rescu",
        org.knowm.xchange.client;uses:="io.github.resilience4j.ratelimiter,io.github.resilience4j.retry,io.vavr.control,jakarta.ws.rs.core,org.knowm.xchange,si.mazi.rescu",
        org.knowm.xchange.currency;uses:="com.fasterxml.jackson.annotation,org.knowm.xchange.instrument",
        org.knowm.xchange.derivative;uses:="com.fasterxml.jackson.annotation,org.knowm.xchange.currency,org.knowm.xchange.instrument",
        org.knowm.xchange.dto;uses:="com.fasterxml.jackson.annotation,org.knowm.xchange.currency,org.knowm.xchange.dto.trade,org.knowm.xchange.instrument",
        org.knowm.xchange.dto.account;
                uses:="com.fasterxml.jackson.annotation,
                        com.fasterxml.jackson.databind.annotation,
                        org.knowm.xchange.currency,
                        org.knowm.xchange.instrument",
        org.knowm.xchange.dto.marketdata;
                uses:="com.fasterxml.jackson.annotation,
                        com.fasterxml.jackson.databind.annotation,
                        org.knowm.xchange.currency,
                        org.knowm.xchange.dto,
                        org.knowm.xchange.dto.trade,
                        org.knowm.xchange.instrument",
        org.knowm.xchange.dto.meta;
                uses:="com.fasterxml.jackson.annotation,
                        com.fasterxml.jackson.core,
                        com.fasterxml.jackson.databind,
                        com.fasterxml.jackson.databind.annotation,
                        org.knowm.xchange.currency,
                        org.knowm.xchange.dto.account,
                        org.knowm.xchange.instrument",
        org.knowm.xchange.dto.trade;
                uses:="com.fasterxml.jackson.annotation,
                        com.fasterxml.jackson.databind.annotation,
                        org.knowm.xchange.currency,
                        org.knowm.xchange.dto,
                        org.knowm.xchange.dto.marketdata,
                        org.knowm.xchange.instrument",
        org.knowm.xchange.exceptions;uses:="org.knowm.xchange.currency,org.knowm.xchange.instrument",
        org.knowm.xchange.instrument;uses:="com.fasterxml.jackson.databind.annotation,org.knowm.xchange.currency,org.knowm.xchange.utils.jackson",
        org.knowm.xchange.interceptor;uses:=si.mazi.rescu,
        org.knowm.xchange.service;
                uses:="io.github.resilience4j.ratelimiter,
                        io.github.resilience4j.retry,
                        javax.crypto,
                        org.knowm.xchange,
                        org.knowm.xchange.client,
                        org.knowm.xchange.dto,
                        org.knowm.xchange.dto.meta,
                        org.knowm.xchange.dto.trade,
                        si.mazi.rescu",
        org.knowm.xchange.service.account;
                uses:="org.knowm.xchange.currency,
                        org.knowm.xchange.dto.account,
                        org.knowm.xchange.instrument,
                        org.knowm.xchange.service,
                        org.knowm.xchange.service.trade.params",
        org.knowm.xchange.service.marketdata;
                uses:="org.knowm.xchange.currency,
                        org.knowm.xchange.dto.marketdata,
                        org.knowm.xchange.instrument,
                        org.knowm.xchange.service,
                        org.knowm.xchange.service.marketdata.params,
                        org.knowm.xchange.service.trade.params",
        org.knowm.xchange.service.marketdata.params;uses:="org.knowm.xchange.currency,org.knowm.xchange.instrument",
        org.knowm.xchange.service.trade;
                uses:="org.knowm.xchange.dto,
                        org.knowm.xchange.dto.account,
                        org.knowm.xchange.dto.trade,
                        org.knowm.xchange.service,
                        org.knowm.xchange.service.trade.params,
                        org.knowm.xchange.service.trade.params.orders",
        org.knowm.xchange.service.trade.params;
                uses:="javax.annotation,
                        org.knowm.xchange.currency,
                        org.knowm.xchange.dto,
                        org.knowm.xchange.dto.account,
                        org.knowm.xchange.instrument",
        org.knowm.xchange.service.trade.params.orders;
                uses:="org.knowm.xchange,
                        org.knowm.xchange.currency,
                        org.knowm.xchange.dto,
                        org.knowm.xchange.dto.trade,
                        org.knowm.xchange.instrument,
                        org.knowm.xchange.service.trade.params",
        org.knowm.xchange.utils;uses:="com.fasterxml.jackson.databind.exc,javax.net.ssl,org.knowm.xchange,org.knowm.xchange.dto,org.knowm.xchange.dto.meta",
        org.knowm.xchange.utils.jackson;uses:="com.fasterxml.jackson.core,com.fasterxml.jackson.databind,org.knowm.xchange.currency,org.knowm.xchange.instrument",
        org.knowm.xchange.utils.nonce;uses:=si.mazi.rescu,
        org.knowm.xchange.utils.timestamp
Import-Package = 
        com.fasterxml.jackson.annotation;resolution:=optional,
        com.fasterxml.jackson.core;resolution:=optional,
        com.fasterxml.jackson.databind;resolution:=optional,
        com.fasterxml.jackson.databind.annotation;resolution:=optional,
        com.fasterxml.jackson.databind.exc;resolution:=optional,

```

Reply:
We should keep the Camel project as the only reference otherwise it will be hard to follow. Keep in mind that Camel only uses a sub-part of the library not 100 % of it. 

But in this particular case, it looks like the Jackson dependencies are provided by `wiremock` with `test` scope which could explain why the tests pass so let's keep it like this